olp-cpp-sdk-dataservice-read/tests/ApiClientLookupTest.cpp

```diff
@@ -103,10 +104,44 @@ TEST(ApiClientLookup, LookupApiSync) {
     Mock::VerifyAndClearExpectations(network.get());
   }
   {
-      // SCOPED_TRACE("Network error propagated to the user");
-  } {
-      // SCOPED_TRACE("Network request cancelled by the user");
-  } {
-    // SCOPED_TRACE("Network error propagated to the user");
+    SCOPED_TRACE("Network error propagated to the user");
+    EXPECT_CALL(*network, Send(IsGetRequest(lookup_url), _, _, _, _))
+        .Times(1)
+        .WillOnce(NetworkMock::ReturnHttpResponse(
+            olp::http::NetworkResponse().WithStatus(
+                olp::http::HttpStatusCode::UNAUTHORIZED),
+            "Inappropriate"));
+
+    client::CancellationContext context;
+    auto response = ApiClientLookup::LookupApi(
+        catalog_hrn, context, service_name, service_version,
+        FetchOptions::OnlineOnly, settings);
+
+    EXPECT_FALSE(response.IsSuccessful());
+    EXPECT_EQ(response.GetError().GetErrorCode(),
+              olp::client::ErrorCode::AccessDenied);
+    Mock::VerifyAndClearExpectations(network.get());
+  }
+  {
+    SCOPED_TRACE("Network request cancelled");
+    EXPECT_CALL(*network, Send(IsGetRequest(lookup_url), _, _, _, _))
+        .Times(1)
+        .WillOnce([=](olp::http::NetworkRequest request,
+                      olp::http::Network::Payload payload,
+                      olp::http::Network::Callback callback,
+                      olp::http::Network::HeaderCallback header_callback,
+                      olp::http::Network::DataCallback data_callback)
+                      -> olp::http::SendOutcome {
+          return olp::http::SendOutcome(olp::http::ErrorCode::CANCELLED_ERROR);
+        });
+    client::CancellationContext context;
+    auto response = ApiClientLookup::LookupApi(
+        catalog_hrn, context, service_name, service_version,
+        FetchOptions::OnlineOnly, settings);
+
+    EXPECT_FALSE(response.IsSuccessful());
+    EXPECT_EQ(response.GetError().GetErrorCode(),
+              olp::client::ErrorCode::Cancelled);
+    Mock::VerifyAndClearExpectations(network.get());
```

Comment:
It's a same test as a previous one, in order to trigger cancel event call context.CancelOperation().
Please ask @diachenko-mischa about cancellation details. For now please remove this test.

Reply:
I guess this is 2 different scenarios:
1. User cancelled operation by itself;
2. The operation was cancelled by network/server.

So, I think the case that you are describing should be covered by separate test.

Reply:
Ok, I see, 2 different types of error propagation.